lib/message_queue/message.ex

```diff
@@ -14,7 +14,7 @@ defmodule MessageQueue.Message do
   @type data() :: term()
   @type opts() :: Access.t()
   @type encoded_message() :: binary() | String.t()
-  @type decoded_message() :: binary() | String.t()
+  @type decoded_message() :: map() | list() | struct()
```

Comment:
actually we can return binary as well here, I think `term()` will be more accurately 
```suggestion
  @type decoded_message() :: term()
```